### PR TITLE
feat(screenshot): init with create-next-app

### PR DESCRIPTION
**Pull request by bestenevoy**

### CODE REVIEW

**Comment by somarlyonks** [comment]
Maybe you'd better have it deployed for preview.

File: src/pages/api/snapshot/desktop/[name].ts [hunk 1]
```diff
@@ -0,0 +1,23 @@
...
+import { genScreenshot } from 'src/helpers'
+
+export interface IQuery {
+    url?: string
+    width?: number
+    height?: number
+}
+
+export default async (req: NextApiRequest, res: NextApiResponse) => {
+    const { url, width, height }= req.query as IQuery
```
Comment (somarlyonks) [inline src/pages/api/snapshot/desktop/[name].ts]: This is unsafe as query are always string. And that's why you transformed parameters in `genScreenshot`. It has to take responsible of its type here.

File: src/pages/api/snapshot/mobile/[name].ts [hunk 2]
```diff
@@ -0,0 +1,20 @@
...
+    height = height ? Number(height) : null
+    if (!url) {
+        res.status(400)
+        return res.end()
+    }
+    if (Array.isArray(url)) {
+        length = url.length
+    }
+    const fileBuffer = await genScreenshot(url, { mode: 'mobile', width, height})
+    res.setHeader("Content-Type", 'jpg')
```
Comment (somarlyonks) [inline src/pages/api/snapshot/mobile/[name].ts]: `image/jpg`

File: src/pages/api/snapshot/mobile/[name].ts [hunk 3]
```diff
@@ -0,0 +1,20 @@
...
+export default async (req: NextApiRequest, res: NextApiResponse) => {
+    let { url, width, height }= req.query as IQuery
+    width = width ? Number(width) : null
+    height = height ? Number(height) : null
+    if (!url) {
+        res.status(400)
+        return res.end()
+    }
+    if (Array.isArray(url)) {
+        length = url.length
```
Comment (somarlyonks) [inline src/pages/api/snapshot/mobile/[name].ts]: looks like dead code

Comment (bestenevoy) [reply src/pages/api/snapshot/mobile/[name].ts]: yes, not yet used

File: src/pages/api/snapshot/mobile/[name].ts (hunk 2 above)
Comment (bestenevoy) [reply src/pages/api/snapshot/mobile/[name].ts]: Get

File: src/helpers/genScreenshot.ts [hunk 4]
```diff
@@ -0,0 +1,62 @@
...
+
+    try {
+        const device = await getDevice()
+        device.viewport.width = options.width || device.viewport.width
+        device.viewport.height = options.height || device.viewport.height
+        let page = await browser.newPage()
+        if (options.mode == 'mobile') {
+            await page.emulate(device)
+        }
+        await page.goto(url)
```
Comment (somarlyonks) [inline src/helpers/genScreenshot.ts]: I believe it's not adequate here. It has to wait some content loading, or even more wait for some basic entry transition. Try theme [a robot named jimmy](https://czg.vercel.app/theme/5486806e35e68e5e001940ff35b11b37) with url `https://czg.vercel.app/theme/5486806e35e68e5e001940ff35b11b37`.

[https://stackoverflow.com/questions/46160929/puppeteer-wait-for-all-images-to-load-then-take-screenshot](https://stackoverflow.com/questions/46160929/puppeteer-wait-for-all-images-to-load-then-take-screenshot) might be helpful.

Comment (somarlyonks) [reply src/helpers/genScreenshot.ts]: @csszen maybe we can merge it first as it works for most circumstances.

**Review by somarlyonks — APPROVED** [review]

**Pull request merged**